src/data_type/multi.value.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+module.exports = class MultiValue {
+  /**
+   * MultiValue constructor.
+   *
+   * @param {array} items
+   */
+  constructor(items) {
+    this.originalItems = items;
+    this.items = items;
+    this.filterInstances = [];
+    this.filterTypes = [];
+  }
+
+  /**
+   * Filter values by their instance type.
+   *
+   * @param {*} type
+   */
+  filterInstance(type) {
+    this.filterInstances.push(type);
+    return this;
+  }
+
+  /**
+   * Filter values by their constructor name.
+   *
+   * @param {String} type
+   */
+  filterType(type) {
+    this.filterTypes.push(type);
+    return this;
+  }
+
+  /**
+   * Apply all filters.
+   */
+  applyFilters() {
+    this.items = this.originalItems;
+
+    if (this.filterInstances.length > 0 || this.filterTypes.length > 0) {
```

Comment:
Can we split this up into `Instance` and `Type` methods? Rather than having an `or`, and doing two different things in one if statement.

Reply:
The reason it's currently combined is to allow either criteria being met:
```javascript
return allowedType || allowedInstance;
```

I'll see if I can achieve this in 2 methods, or at least break these into separate callbacks to make this method smaller.

Reply:
@echarrod I've now split the filters into separate methods in 359dd316e08b30298c805d1de9872af1170f37dd - does this look clearer?